pkgs/tools/misc/beam-book/default.nix

```diff
@@ -0,0 +1,98 @@
+{ lib, stdenv
+, fetchFromGitHub
+, wget, asciidoctor, ditaa, graphviz
+, gitMinimal
+, erlang
+, jdk
+, rsync
+, rouge
+}:
+
+let
+  pname = "the-beam-book";
+  version = "0.0.14.fix";
+
+  bookDirName = "${pname}-${version}";
+
+  book = fetchFromGitHub {
+    name = bookDirName;
+    owner = "happi";
+    repo = "theBeamBook";
+    rev = version;
+    sha256 = "06r19063a9s6hg84garn83rn4cnnkvs58x3j68zvf94mwsmfs86h";
+    leaveDotGit = true;
```

Comment:
We patch the the script anyway. Can't we hardcode the version information?

Reply:
Depends on the patch you are talking about. I'm not sure what version information you want to hardcode.

The `remove-wget.patch` is handled by the other source fetch, `erlangGenOp`. There is a single file from the erlang/otp repository that this book requests, which is normally handled by the makefile running `wget`.

If you are talking about the `add-HEAD-to-git-shortlog.patch`, then that will depend on upstream accepting my change about that.